k9mail-library/src/test/java/com/fsck/k9/mail/store/imap/ImapFolderPusherTest.java

```diff
@@ -0,0 +1,94 @@
+package com.fsck.k9.mail.store.imap;
+
+
+import java.io.IOException;
+
+import android.content.Context;
+
+import com.fsck.k9.mail.K9LibRobolectricTestRunner;
+import com.fsck.k9.mail.MessagingException;
+import com.fsck.k9.mail.PushReceiver;
+import com.fsck.k9.mail.store.StoreConfig;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.robolectric.shadows.ShadowApplication;
+
+import static com.fsck.k9.mail.store.imap.ImapResponseHelper.createImapResponse;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(K9LibRobolectricTestRunner.class)
+public class ImapFolderPusherTest {
+
+    private static final String FOLDER_NAME = "Folder";
+
+    private ImapFolderPusher folderPusher;
+
+    @Mock
+    private ImapStore store;
+    @Mock
+    private ImapFolder folder;
+    @Mock
+    private ImapConnection connection;
+    @Mock
+    private PushReceiver pushReceiver;
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+
+        configurePushReceiver();
+        configureStore();
+        configureFolder();
+
+        folderPusher = new ImapFolderPusher(store, FOLDER_NAME, pushReceiver);
+    }
+
+    @Test
+    public void processStoredUntaggedResponses_withExpungeResponse_shouldTriggerFolderSync() throws Exception {
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                UntaggedHandler untaggedHandler = invocation.getArgumentAt(1, UntaggedHandler.class);
+                untaggedHandler.handleAsyncUntaggedResponse(createImapResponse("* 3 EXPUNGE"));
+                return null;
+            }
+        }).when(folder).executeSimpleCommand(eq(Commands.IDLE), any(UntaggedHandler.class));
+
+        folderPusher.start();
+
+        verify(pushReceiver).syncFolder(FOLDER_NAME);
```

Comment:
@Valodim When I debug this test, this invocation does occur, but this verification still fails. Is there something I am missing here ?